fast64_internal/oot/exporter/scene/actors.py

```diff
@@ -0,0 +1,243 @@
+from dataclasses import dataclass, field
+from typing import Optional
+from mathutils import Matrix
+from bpy.types import Object
+from ....utility import PluginError, CData, indent
+from ...oot_utility import getObjectList
+from ...oot_constants import ootData
+from ...scene.properties import OOTSceneHeaderProperty
+from ..base import Base, Actor
+
+
+@dataclass
+class TransitionActor(Actor):
+    """Defines a Transition Actor"""
+
+    isRoomTransition: Optional[bool] = None
+    roomFrom: Optional[int] = None
+    roomTo: Optional[int] = None
+    cameraFront: Optional[str] = None
+    cameraBack: Optional[str] = None
```

Comment:
I know it's weird but I'd like to keep this initialised to ``None`` so if there's any uncaught/unknown issue it would make it more obvious what the issue is, and just using ``Optional`` since I was told ``thing: str = None`` is bad

Reply:
Wouldn't it be easier to catch errors at construction time of the `TransitionActor` instead of letting this continue on with invalid state?

Reply:
I think I can consider this as solved too (``field(init=False)``), the members are set in the same function as the one that calls this class in the first place (which is ``SceneTransitionActors``'s post init)